extensions/websockets-next/kotlin/src/main/kotlin/io/quarkus/websockets/next/runtime/kotlin/CoroutineInvoker.kt

```diff
@@ -0,0 +1,37 @@
+package io.quarkus.websockets.next.runtime.kotlin
+
+import io.quarkus.arc.Arc
+import io.smallrye.mutiny.Uni
+import io.smallrye.mutiny.coroutines.asUni
+import io.vertx.core.Vertx
+import jakarta.enterprise.invoke.Invoker
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.async
+
+object CoroutineInvoker {
+    @JvmStatic
+    @OptIn(ExperimentalCoroutinesApi::class)
+    fun <T, U> inNewCoroutine(instance: T, arguments: Array<Any?>, invoker: Invoker<T, U>): Uni<U> {
+        val coroutineScope = Arc.container().instance(ApplicationCoroutineScope::class.java).get()
+        val dispatcher: CoroutineDispatcher =
+            Vertx.currentContext()?.let(::VertxDispatcher)
+                ?: throw IllegalStateException("No Vertx context found")
+
+        return coroutineScope
+            .async<U>(context = dispatcher) {
+                suspendCoroutine { continuation ->
+                    arguments[arguments.size - 1] = continuation
```

Comment:
So `Continuation` is always the last parameter of a kotlin suspend method?

Reply:
Yes, that is correct. See also `KotlinContinuationCallbackArgument`.

Reply:
I mean - it's a synthetic param added by kotlin, right?

Reply:
Yes. Except we have to pass it explicitly, because the invoker is not Kotlin-aware.